### PR TITLE
Captcha DoS defense

### PR DESCRIPTION
**Pull request by millerdev**
Add CAPTCHA to forgot password and new account forms

reset-password-captcha

create-account-captcha

@czue @calellowitz

### CODE REVIEW

File: corehq/apps/hqwebapp/templates/hq-captcha-field.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    $('.captcha-refresh').click(function(){
+        var $form = $(this).parent().closest('form');
+        $.getJSON("/captcha/refresh/", {}, function(json) {
+            $form.find('input[name$="captcha_0"]').val(json.key);
+            $form.find('img.captcha').attr('src', json.image_url);
+        });
+        return false;
+    });
+});
+</script>
```
Comment (millerdev) [inline corehq/apps/hqwebapp/templates/hq-captcha-field.html]: Inlined this script to keep things super simple, but could be persuaded to move it to a separate .js file if this seems dirty.

**Comment by calellowitz** [comment]
:+1: looks good to me, test failure was unrelated so i kicked the build again

**Comment by czue** [comment]
@millerdev am I reading it right that this always adds a captcha to those pages? Would it be a huge lift to either 1) only show the captcha after a certain number of requests, or 2) only enable the captcha based on a site-wide setting (`DRACONIAN_SECURITY`?)?

I'm concerned that adding a permanent captcha, especially to the account creation page, could lower our conversion rates. @dimagi/product to weigh in - some version of this is required for the ICDS security audit.

**Comment by czue** [comment]
(:+1: code wise though)

**Comment by amsagoff** [comment]
Thanks for the flag, Cory, I'm also concerned about how this will affect
conversion rates. Either of Cory's proposed workarounds would address my
concerns.

**Comment by millerdev** [comment]
Updated to only show the captcha field when draconian security is enabled. I would prefer to show a captcha when multiple requests have been submitted from a given IP address in some limited time period, but that's more complicated to implement.

**Comment by czue** [comment]
tested locally, looks good

**Pull request merged**